### PR TITLE
[Feature] Path prefix for attachment widget can be defined by expression

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

File: src/gui/editorwidgets/qgsexternalresourceconfigdlg.cpp [hunk 1]
```diff
@@ -36,14 +40,21 @@ QgsExternalResourceConfigDlg::QgsExternalResourceConfigDlg( QgsVectorLayer *vl,
 
   mRootPath->setPlaceholderText( QgsSettings().value( QStringLiteral( "/UI/lastExternalResourceWidgetDefaultPath" ), QDir::toNativeSeparators( QDir::cleanPath( defpath ) ) ).toString() );
 
-  // Add connection to button for choosing default path
-  connect( mRootPathButton, SIGNAL( clicked() ), this, SLOT( chooseDefaultPath() ) );
+  QMenu *rootPathMenu = new QMenu();
+  mRootPathExpressionButton = new QAction( QgsApplication::getThemeIcon( QStringLiteral( "/mIconExpression.svg" ) ), tr( "Expression" ), rootPathMenu );
```
Comment (3nids) [inline src/gui/editorwidgets/qgsexternalresourceconfigdlg.cpp]: icon doesn't seem to show up properly
image

do you confirm?

Comment (m-kuhn) [inline src/gui/editorwidgets/qgsexternalresourceconfigdlg.cpp]: You need to click the arrow and change to an expression.

Comment (3nids) [inline src/gui/editorwidgets/qgsexternalresourceconfigdlg.cpp]: I was thinking about using the data defined button such as in label
image

any reason why not to do so?

Comment (m-kuhn) [inline src/gui/editorwidgets/qgsexternalresourceconfigdlg.cpp]: I wasn't sure how to use the new property framework ;)

**Comment by m-kuhn** [comment]
Manually merged

**Pull request closed without merging**